src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -237,14 +237,10 @@ private static object ToBlittableSupportedType(object root, object value, bool f
             {
                 if (ShouldTreatAsEnumerable(enumerable))
                 {
-                    IEnumerable<object> items;
-
-                    if (value is IEnumerable<object> objectEnumerable)
-                        items = Enumerable.Select(objectEnumerable, x => ToBlittableSupportedType(root, x, flattenArrays, recursiveLevel + 1, engine, context));
-                    else
-                        items = Enumerable.Select(enumerable.Cast<object>(), x => ToBlittableSupportedType(root, x, flattenArrays, recursiveLevel + 1, engine, context));
-
-                    return new DynamicJsonArray(flattenArrays ? Flatten(items) : items);
+                    return value is IEnumerable<object> objectEnumerable
+                        ? EnumerableToJsonArray(flattenArrays ? Flatten(objectEnumerable) : objectEnumerable, root, flattenArrays, recursiveLevel, engine, context)
+                        : EnumerableToJsonArray(flattenArrays ? Flatten(enumerable.Cast<object>()) : enumerable.Cast<object>(), root, flattenArrays, recursiveLevel,
```

Comment:
Can we avoid `.Cast<object>()` ?

Reply:
should I cast each member of enumerable to object in a loop?